clusterloader2/docs/test_suite_design.md

```diff
@@ -0,0 +1,50 @@
+# Cluster loader - TestSuite
+
+## Background
+
+Since 2019 cluster loader become main tool for kubernetes e2e performance tests.
+It is a very convenient tool for running single test case, but not for running
+test bundles. Currently it is impossible to run the same test
+with two different parameter sets in a single job run.
+This forces developers to create separate job configs for almost each test.
+
+## TestSuite
+
+TestSuite is a list of test scenarios running in a single test job.
+Tests will be executed by cluster loader in the order as provided.
+
+```
+type TestSuite []TestScenario
+```
+
+## TestScenario
+
+Test scenario's schema is as following:
+
+```
+type TestScenario struct {
+	// Identifier is a unique test scenario name across test suite.
+	Identifier string
+	// ConfigPath defines path to the config file that satisfy already
+        // existing cluster loader API.
+	ConfigPath string
+	// OverridePaths defines what override files should be applied
+        // to the config specified by the ConfigPath.
+	OverridePaths []string
+}
+```
+
+Assigning overrides on test scenario level has two benefits:
+- Allows to use same template more than once.
+- Allows users to be more flexible with variable naming.
```

Comment:
I don't understand this one.

Reply:
Ohh - it's explained. So should be sth like:
```
Per-scenario overrides benefits:
- ability to use the same template with multiple overrides in a single job
- less error prone - variables with same names across different configs could have been accidentally overwritten before
```

Reply:
Done.